Description: Compile only the classes compatible with the version of jython in Fedora
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/source/build.xml
+++ b/source/build.xml
@@ -275,41 +275,21 @@
         freemarker/ext/jsp/FreeMarkerJspApplicationContext.java"
     />
 
-    <ivy:cachepath conf="build.jython2.0" pathid="ivy.dep.jython2.0" />
+    <ivy:cachepath conf="build.jython2.5" pathid="ivy.dep.jython2.5" />
     <javac srcdir="src/main/java" destdir="build/classes" deprecation="off" 
-      debug="on" optimize="off" target="1.4" source="1.4" encoding="utf-8"
+      debug="on" optimize="off" target="1.5" source="1.5" encoding="utf-8"
       includeantruntime="false"
-      classpathref="ivy.dep.jython2.0"
-      bootclasspath="${boot.classpath.j2se1.4}"
+      classpathref="ivy.dep.jython2.5"
+      bootclasspath="${boot.classpath.j2se1.5}"
       includes="
         freemarker/ext/ant/**,
         freemarker/template/utility/JythonRuntime.java,
         freemarker/ext/jython/**"
       excludes="
-        freemarker/ext/jython/_Jython22VersionAdapter.java,
-        freemarker/ext/jython/_Jython25VersionAdapter.java"
-    />
-    
-    <ivy:cachepath conf="build.jython2.2" pathid="ivy.dep.jython2.2" />
-    <javac srcdir="src/main/java" destdir="build/classes" deprecation="off" 
-      debug="on" optimize="off" target="1.4" source="1.4" encoding="utf-8"
-      includeantruntime="false"
-      classpathref="ivy.dep.jython2.2"
-      bootclasspath="${boot.classpath.j2se1.4}"
-      includes="
+        freemarker/ext/jython/_Jython20And21VersionAdapter.java,
         freemarker/ext/jython/_Jython22VersionAdapter.java"
     />
     
-    <ivy:cachepath conf="build.jython2.5" pathid="ivy.dep.jython2.5" />
-    <javac srcdir="src/main/java" destdir="build/classes" deprecation="off" 
-      debug="on" optimize="off" target="1.4" source="1.4" encoding="utf-8"
-      includeantruntime="false"
-      classpathref="ivy.dep.jython2.5"
-      bootclasspath="${boot.classpath.j2se1.5}"
-      includes="
-        freemarker/ext/jython/_Jython25VersionAdapter.java"
-    />
-    
     <rmic base="build/classes" classpathref="ivy.dep"
       includes="src/main/java/freemarker/debug/impl/Rmi*Impl.class"
       verify="yes" stubversion="1.2"
